### PR TITLE
Bugfix/ctl 991 equals method in service locator service info class returns false positives v2

### PR DESCRIPTION
**Pull request by szogun1987**
Fixes # .

### Checklist

- [ ] I have included examples or tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file
- [x] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Hash is computed in constructor there is no need to delay this operation
- Equals function works more properly 
-

### CODE REVIEW

File: doc/history.txt [hunk 1]
```diff
@@ -69,6 +69,7 @@ Added/fixed:
 (*) CTL-922 Don't cache RootFrame in NavigationService to allow dynamic navigation frames
 (*) CTL-941 Update to json.net 9.x
 (*) CTL-988 Allow view models without model injection to stay alive on DataContext changes
+(x) CTL-991 Equals method in ServiceLocator.ServiceInfo class returns false positives
```
Comment (GeertvanHorrik) [inline doc/history.txt]: Next time group by type, then order by id

File: src/Catel.Core/Catel.Core.Shared/IoC/ServiceLocator.cs [hunk 2]
```diff
@@ -40,39 +40,28 @@ public RegisteredInstanceInfo(ServiceLocatorRegistration registration, object in
...
         {
-            private int _hash;
+            private readonly int _hash;
 
             #region Constructors
             public ServiceInfo(Type type, object tag)
             {
                 Type = type;
                 Tag = tag;
+                _hash = HashHelper.CombineHash(Type.GetHashCode(), Tag != null ? Tag.GetHashCode() : 0);
```
Comment (GeertvanHorrik) [inline src/Catel.Core/Catel.Core.Shared/IoC/ServiceLocator.cs]: Could lazy<t> help here? Or would it be too expensive for garbage collection?

Comment (szogun1987) [inline src/Catel.Core/Catel.Core.Shared/IoC/ServiceLocator.cs]: Main purpose of this class is to be dictionary key. I don't know if it is created without being key. I think that lazy computation of hash-code makes this code more complex and is always delayed never prevented.

File: doc/history.txt (hunk 1 above)
Comment (szogun1987) [inline doc/history.txt]: Fixed

**Comment by GeertvanHorrik** [comment]
Thanks!

**Pull request merged**